build_system/src/config.rs

```diff
@@ -299,10 +311,11 @@ impl ConfigInfo {
     ) -> Result<(), String> {
         env.insert("CARGO_INCREMENTAL".to_string(), "0".to_string());
 
-        if self.gcc_path.is_empty() && !use_system_gcc {
+        if self.gcc_path.is_none() && !use_system_gcc {
             self.setup_gcc_path()?;
         }
-        env.insert("GCC_PATH".to_string(), self.gcc_path.clone());
+        let gcc_path = self.gcc_path.clone().expect("GCC path wasn't provided");
```

Comment:
Just to make sure, since it's still not clear to me, this `expect()` is never supposed to fail because we check for errors in `config.rs`, right?
If so, please change to something like this:
```suggestion
        let gcc_path = self.gcc_path.clone().expect("The config module should have emitted an error if the GCC path wasn't provided");
```